### DISCUSSION TITLE
Asking feature of framesForwardKinematics function and example of static_contact_dynamics

### DISCUSSION THREAD

**Discussion by JuJuManCheng** [stack-of-tasks/pinocchio#2391] [Q&A]
Hello, I'm working on a porject trying to get the GRF according to dynamics model. However, the example in the source code is the standing phase. I'd like to ask if this dynamic model also works for swinging phase, such as while the quadruped robot is walking?

Besides, I know the framesForwardKinematics function can get the position and rotation matrix of each joints of the quadruped. I would like to ask if it's possible to obtain the foot position of each legs via this function. Further, is the rotation matrix the relationship bewtween the joint frame and the baselink frame? or it's the relationship between the joint frame and the previous joint frame?( e.g: the rotation we got from one of the calf joint, is it the rotation matrix between (foot, calf), (calf, thigh) or (calf, baselink)? )

**Comment by jorisv** [comment]
Hello,

What GRF stand for ?
Dynamic model is working with any set of contacts, but you have to configure it with the right contact model.

Some URDF models contains links to obtains the foot position. It can be retrieved in pinocchio as a frame. Here an example for Talos:

```python
feet_name = ["left_sole_link", "right_sole_link"]
frame_ids = [model.getFrameId(frame_name) for frame_name in feet_name]
```

In the `Data` structure, you have the oMf vector that contains transformation from the world origin to each robot frame.
